doc/index.rst

```diff
@@ -207,26 +225,76 @@ Pulling Docker image from an online hub
+
+(Gitlab CI) Access to private repositories
+------------------------------------------
 
-If your Docker image is missing any of the above libraries, then you can still pass their name by `ADDITIONAL_DEBS` (see `variables section <./index.rst#optional-environment-variables>`_).
+To open PRs/MRs (merge requests, that's what Gitlab calls pull requests) that access private Gitlab repos, additional setting is needed both on:
 
-Note-1. This disables the handling of `ROS_REPOSITORY_PATH` and `ROS_DISTRO` as ROS needs already to be installed in the image.
+- Your repo (i.e. where PRs/MRs targets): Add ssh private keys. Add in the CI settings.
+- The private repos your PRs/MRs access: Matching public keys must be set as `Deploy Key`.
 
-Note-2. For some images, `ROS_DISTRO` variable still needs to be set. This holds for `ROS official Docker images <https://hub.docker.com/_/ros/>`_ as of Sept. 2017.
+#. If you haven't done so, create SSH key pair (`reference on gitlab.com <https://docs.gitlab.com/ce/ssh/README.html#generating-a-new-ssh-key-pair>`_).
+#. Open your fork with your account filled in in the repo's URLs below (the rest of the steps need to be done for these URLs). `%YOUR_GITLAB_DOMAIN%` can be `gitlab.com` if you're using the hosted version of Gitlab.
 
-(Optional but recommended) Subscribe to the change in this repo (industrial_ci)
+   #. https://%YOUR_GITLAB_DOMAIN%/%YOUR_ACCOUNT%/%REPO_YOUR_BRANCH_RESIDES%/settings/ci_cd
+   #. https://gitlab.com/130s/sample_gitlab_private/settings/ci_cd  (concrete example)
+
+#. Expand "`Secret variables`" section.
+#. In "Add a variable" section, fill in the following text field/area.
+
+   #. **Key**: `SSH_PRIVATE_KEY`
+   #. **Value**: Copy paste the entire content of your private key file.
+
+     #. Include the header and footer, i.e.  `-----BEGIN/END RSA PRIVATE KEY-----`.
+#. In "Add a variable" section again, fill in the following text field/area.
+
+   #. **Key**: `SSH_SERVER_HOSTKEYS`
+   #. **Value**: Copy paste the entire line of the following: On your Linux computer, run `ssh-keyscan gitlab.com`. You should get a hash key entry/ies. Copy the entire line that is NOT commented out. For example, the author gets the following, and copied the 2nd line (, which may render as separate lines on your web browser, but it's a long single line):
+
+     ::
+
+      # gitlab.com:22 SSH-2.0-OpenSSH_7.2p2 Ubuntu-4ubuntu2.2
+      gitlab.com ssh-rsa RandomKeySequenceRandomKeySequenceRandomKeySequenceRandomKeySequenceRandomKeySequenceRandomKeySequenceRandomKeySequence
+      # gitlab.com:22 SSH-2.0-OpenSSH_7.2p2 Ubuntu-4ubuntu2.2
+      gitlab.com ecdsa-sha2-nistp256 RandomKeySequenceRandomKeySequenceRandomKeySequenceRandomKeySequenceRandomKeySequenceRandomKeySequenceRandomKeySequence
+      # gitlab.com:22 SSH-2.0-OpenSSH_7.2p2 Ubuntu-4ubuntu2.2
+
+#. (Optional) If you're an admin of the private repos, add a public key (`refrence <https://docs.gitlab.com/ce/ssh/README.html#deploy-keys>`_).
```

Comment:
This is not optional. If the private repositories your CI jobs need to access don't have the matching public key as deploy key (or somehow grant access to the private key you set up in your repo), all of the above won't work.